ynr/settings/base.py

```diff
@@ -345,19 +345,38 @@ def root(*path):
     }
 }
 
-Q_CLUSTER = {
-    "name": "DjangORM",
-    "workers": 4,
-    "queue_limit": 50,
-    "bulk": 10,
-    "orm": "default",
-    # schedule-friendly settings
-    # TODO: review once we have proper background tasks
-    "catch_up": False,
-    "timeout": 240,
-    "max_attempts": 1,  # no retries
-    "retry": 300,  # irrelevant, but this must be a number greater than timeout
-}
+
+def get_qcluster_settings(minimum_cron_interval_seconds):
+    # kill jobs if they haven't completed in $TIMEOUT seconds
+    TIMEOUT = 240
+    # retry must be a number greater than timeout
+    # and less than minimum_cron_interval_seconds.
+    # We can't't have scheduled tasks that run
+    # more frequently than every $RETRY seconds
+    RETRY = TIMEOUT + (minimum_cron_interval_seconds - TIMEOUT) / 2
+
+    assert TIMEOUT < RETRY < minimum_cron_interval_seconds
```

Comment:
I've done two things here:

Firstly, I've set these settings to better values. Related reading: https://django-q2.readthedocs.io/en/master/configure.html#retry
Secondly, I've tried to implement a rule here that makes it harder for us to set them to silly values
This rule corresponds to my current understanding of how to set these, but we might discover in future that this is also bad
This is probably going to change again when we start doing on-demand tasks